### PR TITLE
Fix CAN_O_NOSleep and now all ATTR Decay item will never sleep

### PR DESCRIPTION
**Pull request by Jhobean**
Fix Issue #956

### CODE REVIEW

**Comment by AtillaG1** [comment]
When will this be commited ? 😍😎

File: src/game/CSector.cpp [hunk 1]
```diff
@@ -191,15 +191,15 @@ void CSector::_GoSleep()
 		CChar* pChar = static_cast<CChar*>(pObjRec);
 		const bool fSleeping = pChar->IsSleeping();
 		ASSERT(pChar->IsDisconnected());
-		if (!fSleeping)
+		if (!fSleeping && !pChar->Can(CAN_O_NOSLEEP))
```
Comment (cbnolok) [inline src/game/CSector.cpp:194]: Can't those checks be moved in the cantick method?

File: src/game/items/CItem.cpp [hunk 2]
```diff
@@ -5964,8 +5964,13 @@ bool CItem::_OnTick()
     EXC_SET_BLOCK("sleep check");
 
     const CSector* pSector = GetTopSector();	// It prints an error if it belongs to an invalid sector.
-    if (pSector && pSector->IsSleeping())
+    if (pSector && pSector->IsSleeping() && !Can(CAN_O_NOSLEEP))
```
Comment (cbnolok) [inline src/game/items/CItem.cpp:5967]: Same thing here for the checks for nosleep and decay

**Comment by Jhobean** [comment]
I need you help @cbnolok. Can tick method just not work on sleeping sector.

**Comment by cbnolok** [comment]
Solved by the last commit

**Pull request closed without merging**